docs/guides/private-import.md

```diff
@@ -2,47 +2,76 @@
-
-```python
-from owid.walden import Dataset
-
-local_file = 'private_test.csv'
-metadata = {
-  'name': 'private_test',
-  'short_name': 'private_test',
-  'description': 'testing private walden data',
-  'source_name': 'test',
-  'url': 'test',
-  'license_url': 'test',
-  'date_accessed': '2022-02-07',
-  'file_extension': 'csv',
-  'namespace': 'private',
-  'publication_year': 2021,
-}
-
-# upload the local file to Walden's cache
-dataset = Dataset.copy_and_create(local_file, metadata)
-# upload it as private file to S3
-url = dataset.upload(public=False)
-# update PUBLIC walden index with metadata
-dataset.save()
+
+While most of the data at OWID is publicly available, some datasets are added to our catalogue with some restrictions. These include datasets that are not redistributable, or that are not meant to be shared with the public. This can happen due to a strict license by the data provider, or because the data is still in a draft stage and not ready for public consumption.
+
+Various privacy configurations are available:
+
+- Skip re-publishing to GitHub.
+- Disable data downloading options on Grapher.
+- Disable public access to the original file (snapshot).
+- Hide the dataset from our public catalog (accessible vua `owid-catalog-py`).
+
+In the following, we explain how to create private steps in the ETL pipeline and how to run them.
+
+## Create a private step
+
+### Snapshot
+
+To create a private snapshot step, set the `meta.is_public` property in the snapshot .dvc file to false:
+
+```yaml
+meta:
+  is_public: false
+
+  origin:
+    # Data product / Snapshot
+    title: World Population Prospects
+    ...
+```
+
+This will prevent the file to be publicly accessible without the appropriate credentials.
+
+### Meadow, Garden, Grapher
+
+Creating a private data step means that the data is not listed in the public catalog. Also, private datasets will not be re-published to GitHub.
+
+To create a private data step (meadow, garden or grapher) simply use `data-private` prefix in the step name in the DAG. For example, the step `grapher/ihme_gbd/2024-06-10/leading_causes_deaths` (this is from [health.yml](https://github.com/owid/etl/blob/master/dag/health.yml)) is private:
+
+```yaml
+# IHME GBD Leading cause of  deaths - update
+data-private://meadow/ihme_gbd/2024-06-10/cause_hierarchy:
+  - snapshot-private://ihme_gbd/2024-06-10/cause_hierarchy.csv
+data-private://garden/ihme_gbd/2024-06-10/leading_causes_deaths:
+  - data-private://garden/ihme_gbd/2024-05-20/gbd_cause
+  - data-private://meadow/ihme_gbd/2024-06-10/cause_hierarchy
+data-private://grapher/ihme_gbd/2024-06-10/leading_causes_deaths:
+  - data-private://garden/ihme_gbd/2024-06-10/leading_causes_deaths
 ```
 
+!!! note "Make the data non-downloadable"
+
+    To make the data non-downloadable on Grapher, set the `non_redistributable` property in the dataset metadata to `true`:
```

Comment:
Maybe it's worth mentioning that this is usually done in the garden metadata file.